developer_docs/how_to_create_a_plugin.asciidoc

```diff
@@ -1750,11 +1750,21 @@ If you want the setting to have initial value please use migration to seed it.
 ```
 class SeedInitialFooSettingValue < ActiveRecord::Migration[6.0]
   def up
-    Setting.find_or_create(name: 'foo').update(value: Foreman.uuid)
+    Setting.find_or_create_by(name: 'foo', category: 'Setting').update(value: Foreman.uuid)
   end
 end
 ```
```

Comment:
technically we could drop that suggestion now, but I wasn't sure we should just now.

Reply:
I think we should. Having bad recommendations is a recipe for bad code. I recall seeing some paper that investigated bugs introduced in code examples and tracing them back to real life code.

Reply:
I'd still love to hear from @ezr-ondrej why the original suggestion was a migration over a seed, tbh.

Reply:
Sorry for missing this.

I think it was mainly due to worries for stability of the value through HA setups. Which is only an issue with randomized initial values, so basically only the instance_id is a real issue I know of (which might be just wrong to use settings for and not something we want to document for other cases :) ).

Why migration over seed - I believe it was because prior the whole setting refactoring out of DB, the setting would get created prior seeds get to run, so we couldn't use `find_or_create` in seeds (tho probably we could fill out empty value and that would have been fine, tho there might be some non empty contraint preventing it). **Now seeds should be safe to use IMHO**, as the settings do not get autocreated anymore.

In general I'd just not advertize posibility for initial value as that sounds like a wrong use of settings ^_^